src/settings/menuoptions.cpp

```diff
@@ -0,0 +1,93 @@
+#include "menuoptions.h"
+#include <QDebug>
+
+#include <QVBoxLayout>
+MenuOptions::MenuOptions(QWidget *pParent) : QDialog (pParent)
+{
+    setModal(true);
+    int width = 850;
+    int height = 280;
+    setMinimumWidth(width);
+    setMaximumWidth(width);
+    setMinimumHeight(height);
+    setMaximumHeight(height);
+    setWindowTitle("Settings");
+
+    tabWidget = new QTabWidget;
+
+    tabEnvironment = new TabEnvironment(this);
+    tabTextEditor = new TabTextEditor(this);
+    tabCpp = new TabCpp(this);
+    tabRepository = new TabRepository(this);
+    tabConnection = new TabConnection(this);
+
+
+    tabWidget->addTab(tabEnvironment, "Environment");
+    tabWidget->addTab(tabTextEditor, "Text Editor");
+    tabWidget->addTab(tabCpp, "C++");
+    tabWidget->addTab(tabRepository, "Repository");
+    tabWidget->addTab(tabConnection, "Connection");
+
+    buttonBox = new QDialogButtonBox(QDialogButtonBox::Ok
+           | QDialogButtonBox::Apply | QDialogButtonBox::Cancel);
+
+    connect(buttonBox, &QDialogButtonBox::clicked, this, &MenuOptions::onBtnClicked);
+
+    QVBoxLayout *mainLayout = new QVBoxLayout;
+    mainLayout->addWidget(tabWidget);
+    mainLayout->addWidget(buttonBox);
+    setLayout(mainLayout);
+    show();
+}
+
+void MenuOptions::onBtnApplyClicked()
+{
+    //      APPLY ALL CHANGES IN TABS
+    // tab text editor
+    if (tabTextEditor->newValuesToChange.size())
+    {
+        emit valuesForEditor(tabTextEditor->newValuesToChange);
```

Comment:
here you emit the signal instantaneous. But in the code below you're doing a lot of checks for tabTextEditor->newValuesToChange. Following the logic of this code, firstly you emit the signal and after that you check is this value that you used to emit correct or not. Explain please.